clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -683,21 +686,65 @@ public Node provide() {
         }
     }
 
+    /**
+     * Provides context which the retry may refer to
+     */
+    class CallRetryContext {
+
+        private int tries = 0;
+        private long nextAllowedTryMs = 0;
+        private final double JITTER_MIN = 0.8;
+        private final double JITTER_MAX = 1.2;
+
+        public int tries() {
+            return tries;
+        }
+
+        public long nextAllowedTryMs() {
+            return nextAllowedTryMs;
+        }
+
+        private void updateTries(int currentTries) {
+            this.tries = currentTries + 1;
+        }
+
+        private void updateNextAllowTryMs() {
+            double jitter = Math.random() * (JITTER_MAX - JITTER_MIN) + JITTER_MIN;
+            int failures = tries - 1;
+            double exp = Math.pow(2, failures);
+            this.nextAllowedTryMs = time.milliseconds() +
+                    (long) Math.min(retryBackoffMaxMs, jitter * exp * retryBackoffMs);
+            // TODO: Remove the line below
+            System.out.println("nextAllow = " +  (long) Math.min(retryBackoffMaxMs, jitter * exp * retryBackoffMs));
+        }
+
+        /**
+         * Update the current context upon the failed call's context
+         *
+         * @param failedCallRetryContext    The failed call's retry context
+         */
+        public void update(CallRetryContext failedCallRetryContext) {
```

Comment:
Let's rename the method to be more explicit about what we're updating here.
```suggestion
        public void updateCallRetryContext(CallRetryContext failedCallRetryContext) {
```

Reply:
Refactored.

Reply:
Seems as if this refactor hasn't been updated on the PR